rpc/src/main/scala/proto/converters.scala

```diff
@@ -0,0 +1,194 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package freestyle
+package rpc
+package protocol
+
+import freestyle.rpc.protocol.model._
+
+import scala.meta.Defn.{Class, Object, Trait}
+import scala.meta._
+import scala.meta.contrib._
+
+object converters {
+
+  trait ScalaMetaSource2ProtoDefinitions {
+    def convert(s: Source): ProtoDefinitions
+  }
+
+  class DefaultScalaMetaSource2ProtoDefinitions(
+      implicit OC: ScalaMetaObject2ProtoOptions,
+      MC: ScalaMetaClass2ProtoMessage,
+      SC: ScalaMetaTrait2ProtoService)
+      extends ScalaMetaSource2ProtoDefinitions {
+
+    override def convert(s: Source): ProtoDefinitions = ProtoDefinitions(
+      options = optionDirectives(s).toList.flatMap(OC.convert),
+      messages = messageClasses(s).map(MC.convert).toList,
+      services = serviceClasses(s).map(SC.convert).toList
+    )
+
+    private[this] def optionDirectives(source: Source): Seq[Object] = source.collect {
+      case o: Object /*if o.hasMod(mod"@option")*/ => o
+    }
+
+    private[this] def messageClasses(source: Source): Seq[Class] = source.collect {
+      case c: Class if c.hasMod(mod"@message") => c
+    }
+
+    private[this] def serviceClasses(source: Source): Seq[Trait] = source.collect {
+      case t: Trait if t.hasMod(mod"@service") => t
+    }
+  }
+
+  object ScalaMetaSource2ProtoDefinitions {
+    implicit def defaultSourceToProtoDefinitions(
+        implicit OC: ScalaMetaObject2ProtoOptions,
+        MC: ScalaMetaClass2ProtoMessage,
+        SC: ScalaMetaTrait2ProtoService): ScalaMetaSource2ProtoDefinitions =
+      new DefaultScalaMetaSource2ProtoDefinitions
+  }
+
+  trait ScalaMetaClass2ProtoMessage {
+    def convert(c: Class): ProtoMessage
+  }
+
+  object ScalaMetaClass2ProtoMessage {
+    implicit def defaultClass2MessageConverter(implicit PC: ScalaMetaParam2ProtoMessageField) =
+      new ScalaMetaClass2ProtoMessage {
+        override def convert(c: Class): ProtoMessage = ProtoMessage(
+          name = c.name.value,
+          fields =
+            c.ctor.paramss.flatten.zipWithIndex.map { case (p, t) => PC.convert(p, t + 1) }.toList
+        )
+      }
+  }
+
+  trait ScalaMetaParam2ProtoMessageField {
+    def convert(p: Term.Param, tag: Int): ProtoMessageField
+  }
+
+  object ScalaMetaParam2ProtoMessageField {
+    implicit def defaultParam2ProtoMessageField: ScalaMetaParam2ProtoMessageField =
+      new ScalaMetaParam2ProtoMessageField {
```

Comment:
It could be just:

```scala
object ScalaMetaParam2ProtoMessageField {
    implicit def defaultParam2ProtoMessageField: ScalaMetaParam2ProtoMessageField =
      (p: Term.Param, tag: Int) => p match {
        case param"..$mods $paramname: Double = $expropt" =>
          ProtoDouble(name = paramname.value, tag = tag)
             ...
```